docs/isp_flashing_guide.md

```diff
@@ -1,263 +1,242 @@
+|[SparkFun Pro Micro (5V/16MHz)](https://github.com/sparkfun/Arduino_Boards/blob/master/sparkfun/avr/bootloaders/caterina/Caterina-promicro16.hex)                |`0xFF`|`0xD8`|`0xFB`  |`1B4F:9205`|
+|[SparkFun LilyPadUSB (and some Pro Micro clones)](https://github.com/sparkfun/Arduino_Boards/blob/main/sparkfun/avr/bootloaders/caterina/Caterina-lilypadusb.hex)|`0xFF`|`0xD8`|`0xFE`  |`1B4F:9207`|
+|[Pololu A-Star 32U4](https://github.com/pololu/a-star/blob/master/bootloaders/caterina/Caterina-A-Star.hex)*                                                     |`0xFF`|`0xD0`|`0xF8`  |`1FFB:0101`|
+|[Adafruit Feather 32U4](https://github.com/adafruit/Caterina-Bootloader/blob/master/Built%20Firmwares/Caterina-Feather32u4.hex)                                  |`0xFF`|`0xD8`|`0xFB`  |`239A:000C`|
+|[Adafruit ItsyBitsy 32U4 (3V3/8MHz)](https://github.com/adafruit/Caterina-Bootloader/blob/master/Caterina_itsybitsy3V.hex)*                                      |`0xFF`|`0xD8`|`0xFB`  |`239A:000D`|
+|[Adafruit ItsyBitsy 32U4 (5V/16MHz)](https://github.com/adafruit/Caterina-Bootloader/blob/master/Caterina_itsybitsy5V.hex)                                       |`0xFF`|`0xD8`|`0xFB`  |`239A:000E`|
+|[Arduino Leonardo](https://github.com/arduino/ArduinoCore-avr/blob/master/bootloaders/caterina/Caterina-Leonardo.hex)*                                           |`0xFF`|`0xD8`|`0xFB`  |`2341:0036`|
+|[Arduino Micro](https://github.com/arduino/ArduinoCore-avr/blob/master/bootloaders/caterina/Caterina-Micro.hex)*                                                 |`0xFF`|`0xD8`|`0xFB`  |`2341:0037`|
 
-Open a terminal (`cmd` on Windows, for instance) and navigate to your where your modified .hex file is. We'll pretend this file is called `main.hex`, and that your Teensy 2.0 is on the `COM3` port - if you're unsure, you can open your Device Manager, and look for `Ports > USB Serial Device`. Use that COM port here. You can confirm it's the right port with:
+?> Files marked with a * have combined Arduino sketches, which runs by default and also appears as a serial port. However, this is *not* the bootloader device.
 
-    avrdude -c avrisp -P COM3 -p atmega32u4
+### BootloadHID (PS2AVRGB)
 
-and you should get something like the following output:
+This bootloader is primarily for keyboards originally designed for the PS2AVRGB firmware and Bootmapper Client. It is not recommended for use in new designs.
 
-    avrdude: AVR device initialized and ready to accept instructions
+|MCU                                                                                                        |Low   |High  |USB ID     |
+|-----------------------------------------------------------------------------------------------------------|------|------|-----------|
+|[ATmega32A](https://github.com/qmk/qmk_firmware/blob/master/util/bootloader_ps2avrgb_bootloadhid_1.0.1.hex)|`0x0F`|`0xD0`|`16C0:05DF`|
```

Comment:
Where do these fuse values come from? Looks like they set:
* `CKSEL[3:0] = 1111`
* `SUT[1:0] = 00`
* `CKOPT = 1`

Apparently with `CKOPT = 1` and `CKSEL[3:1] = 111` the documented frequency range is 3.0–8.0 MHz, and the `CKSEL[0] = 1`, `SUT[1:0] = 00` combination sets the start-up time to 1K CK + 65 ms, which is documented as “Ceramic resonator, slowly rising power”, with a footnote that this option “ can also be used with crystals when not operating close to the maximum frequency of the device, and if frequency stability at start-up is not important for the application”.

The USBaspLoader settings for the same ATmega32A chip below (`L=0x1F, H=0xC0`) differ only in the clock setting part:
* `CKSEL[3:0] = 1111`
* `SUT[1:0] = 01`
* `CKOPT = 0`

In this case `CKOPT = 0` select the full-swing mode for the generator, which is documented to have a “wide frequency range” (up to 16 MHz for ceramic resonators, no maximum mentioned in that section for crystal oscillators), and the `CKSEL[0] = 1`, `SUT[1:0] = 01` combination sets the start-up time to 16K CK, which is documented as “Crystal Oscillator, BOD enabled”.

However, if the `L=0x0F, H=0xD0` settings work in practice, maybe they can just be left as is.

Reply:
These values come from the Winkeyless blog: https://blog.winkeyless.kr/16
And to make sure, I made my JJ4x4 (which I know I have never changed the fuses on as it doesn't have a header) spit out the fuse values over console, and they match up.

However, these values aren't any different to `L=0x1F, H=0xC0`, because the bits that differ on the 32A are undefined. The engbedded calculator also seems to detect this and adds an additional note in the "AVRDUDE arguments" section.

Reply:
Hmm, according to the datasheet, ATmega32A does not have any undefined bits in its just two fuse bytes. But if those values always worked, maybe keeping them as is would be OK.

Reply:
I guess the calculator(s) have some inconsistencies, as those bits are not available to be toggled individually but the top section shows the CKOPT and SUT values...